File: de.tototec.tools.cmvn/src/main/java/de/tototec/tools/cmvn/model/EclipseClasspathGeneratorConfig.java
package de.tototec.tools.cmvn.model;

import java.util.LinkedList;
import java.util.List;
import java.util.Map;

import lombok.Data;

@Data
public class EclipseClasspathGeneratorConfig {
	private boolean autoGenerateCompile = false;
	private boolean autoGenerateTest = false;
	private boolean autoGenerateRuntime = false;
	private List<Map<String, String>> cpEntries = new LinkedList<Map<String, String>>();
}

File: de.tototec.cmvn/src/main/scala/de/tototec/cmvn/model/EclipseClasspathGeneratorConfig.scala
package de.tototec.cmvn.model

import java.util.LinkedList
import java.util.List
import java.util.Map
import lombok.Data
//remove if not needed
import scala.collection.JavaConversions._

case class EclipseClasspathGeneratorConfig(var autoGenerateCompile: Boolean = false,
                                           var autoGenerateTest: Boolean = false,
                                           var autoGenerateRuntime: Boolean = false,
                                           var cpEntries: List[Map[String, String]] = new LinkedList[Map[String, String]]()) {

  def this() = this(autoGenerateCompile = false)

}
